### PR TITLE
SISRP-26566 - Exclude unknown Advisor types for Graduate students

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.berkeley.edu/browse/SISRP-26566

### CODE REVIEW

File: app/models/advising/my_advising.rb [hunk 1]
```diff
@@ -107,5 +110,11 @@ def transform_date(item, key)
       item[key] = format_date Time.zone.parse(item[key]).to_datetime
     end
 
+    def student_is_graduate?
+    	college_and_level = MyAcademics::CollegeAndLevel.new(@uid)
+      plans = college_and_level.hub_college_and_level[:plans]
+      career_codes = plans.to_a.collect {|plan| plan[:career].try(:[], :code) }
+      career_codes.include?("GRAD")
+    end
```
Comment (lyttam) [inline app/models/advising/my_advising.rb]: @redconfetti indentation on line 114 and single quotes on line 117 (if it matters).

File: app/models/advising/my_advising.rb [hunk 2]
```diff
@@ -65,7 +65,10 @@ def sort_advisors(advisors)
         matched_advisors = grouped_advisors.delete(type).to_a
         sorted_advisors.concat(matched_advisors)
       end
-      remaining_advisors = grouped_advisors.values.flatten
+      remaining_advisors = []
+      unless student_is_graduate?
+        remaining_advisors = grouped_advisors.values.flatten
+      end
       sorted_advisors.concat(remaining_advisors)
```
Comment (lyttam) [inline app/models/advising/my_advising.rb]: @redconfetti You could avoid doing this `concat` operation when unnecessary if you moved it inside the `unless` block.

Comment (lyttam) [inline app/models/advising/my_advising.rb]: @redconfetti You could avoid doing this `concat` operation when unnecessary if you moved it inside the `unless` block.

**Comment by lyttam** [comment]
@redconfetti I looked at your Jira but it was a lot of words.  Can you help me understand the business reason for adding additional advisors to the list for non-GRAD students?

File: app/models/advising/my_advising.rb (hunk 2 above)
Comment (redconfetti) [inline app/models/advising/my_advising.rb]: @lyttam - LOL, you're right. 😺

File: app/models/advising/my_advising.rb (hunk 1 above)
Comment (redconfetti) [inline app/models/advising/my_advising.rb]: Doh!

**Comment by raydavis** [comment]
@lyttam , you OK with the review?

**Pull request merged**